ap/tasks/reduction.py

```diff
@@ -220,28 +240,91 @@ def get_avg_std(values):
-
-    sandbox = dev_sandbox("bash::$AP_BASE/sandboxes/venv_columnar.sh")
-
-    shifts = {SelectEvents}
+class MergeReducedEventsUser(DatasetTask):
 
     # recursively merge 8 files into one
     merge_factor = 8
 
-    # the key of the config that defines the merging in the branch map of DatasetTask
-    file_merging = "after_reduction"
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # cached value of the file_merging until it's positive
+        self._cached_file_merging = -1
+
+        # in case this is a workflow, do not cache branches by default
+        # (this is enabled in reduced_file_merging once positive)
+        self._cache_branches = False
+
+    @property
+    def file_merging(self):
+        """
+        Needed by DatasetTask to define the default branch map.
+        """
+        if self._cached_file_merging < 0:
+            # reset the forest in case this is a forest ForestMerge task
+            self._forest_built = False
+
+            # check of the merging stats is present and of so, set the cached file merging value
+            output = MergeReductionStats.req(self).output()
+            if output.exists():
+                self._cached_file_merging = output.load(formatter="json")["merge_factor"]
+                self._cache_branches = True
+
+        return self._cached_file_merging
+
+    @property
+    def merging_stats_exist(self):
+        return self.file_merging >= 1
+
+    def reduced_dummy_output(self):
+        # dummy output to be returned in case the merging stats are not present yet
+        return self.local_target("DUMMY_UNTIL_REDUCED_MERGING_STATS_EXIST")
+
+    @classmethod
+    def maybe_dummy(cls, func):
+        # meant to wrap output methods of tasks depending on merging stats
+        # to inject a dummy output in case the status are not there yet
+        @functools.wraps(func)
+        def wrapper(self):
+            # when the merging stats do not exist yet, return a dummy target
+            if not self.merging_stats_exist:
+                return self.reduced_dummy_output()
+
+            # otherwise, bind the wrapped function and call it
+            return func.__get__(self, self.__class__)()
+
+        return wrapper
+
+
+class MergeReducedEvents(MergeReducedEventsUser, CalibratorsSelectorMixin, law.tasks.ForestMerge, HTCondorWorkflow):
+
+    sandbox = dev_sandbox("bash::$AP_BASE/sandboxes/venv_columnar.sh")
+
+    shifts = {ReduceEvents, MergeReductionStats}
 
     def create_branch_map(self):
         # DatasetTask implements a custom branch map, but we want to use the one in ForestMerge
         return law.tasks.ForestMerge.create_branch_map(self)
 
     def merge_workflow_requires(self):
-        return ReduceEvents.req(self, _exclude={"branches"})
+        return {
+            "stats": MergeReductionStats.req(self),
+            "events": ReduceEvents.req(self, _exclude={"branches"}),
```

Comment:
(Just to stress this, I'm just asking to make sure :D )

Do we have to also exclude the singular here? So
```suggestion
            "events": ReduceEvents.req(self, _exclude={"branch", "branches"}),

Reply:
`merge_workflow_requires` is only called by workflows, whose branch value is anyway -1, hence it's actually wanted that the value is passed on via req do denote that "this workflow requires a workflow itself."